configure.py

```diff
@@ -135,9 +135,15 @@ def try_compile_and_link(compiler, source='', flags=[]):
             sfile.file.flush()
             # We can't write to /dev/null, since in some cases (-ftest-coverage) gcc will create an auxiliary
             # output file based on the name of the output file, and "/dev/null.gcsa" is not a good name
-            return subprocess.call([compiler, '-x', 'c++', '-o', ofile, sfile.name] + args.user_cflags.split() + flags,
-                                   stdout=subprocess.DEVNULL,
-                                   stderr=subprocess.DEVNULL) == 0
+            if verbose:
+                ret = subprocess.run([compiler, '-x', 'c++', '-o', ofile, sfile.name] + args.user_cflags.split() + flags,
+                                    capture_output=True)
+                print(ret.stderr.decode('utf-8'))
+            else:
+                ret = subprocess.run([compiler, '-x', 'c++', '-o', ofile, sfile.name] + args.user_cflags.split() + flags,
+                                    stdout=subprocess.DEVNULL,
+                                    stderr=subprocess.DEVNULL)
```

Comment:
Why the duplication? Apart from the standard "you have to modify in two places", the different parameters may themselves affect the result of the test. Shouldn't you put just the print() under the conditional?

Reply:
Not really, it's not duplication - the first call uses `capture_output=True`, and the second one pushes everything to /dev/null. But I forgot to move the comment above inside the branch, so I'll do it in v2.

Reply:
I meant that you can capture_output unconditionally, and print it conditionally. So there'd be just one subprocess.run() call.

Reply:
Ah, sure, for some reason I optimized it as if it was on hot path, which it's actually very far from (first clue: the `.py` extension). I'll send v3 tomorrow